services/user.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: Optional[str] = None,
+    first_name: Optional[str] = None,
+    last_name: Optional[str] = None,
+) -> Optional[User]:
+    user = User.objects.create_user(
+        username=username,
+        password=password,
+        email=email,
+    )
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> Optional[User]:
+    return User.objects.get(id=user_id)
+
+
+def update_user(
+    user_id: Optional[int],
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+    email: Optional[str] = None,
+    first_name: Optional[str] = None,
+    last_name: Optional[str] = None,
+) -> None:
+    user = get_user(user_id)
```

Comment:
if user does not exist then an exception will be triggered so basically here you will always have a user instance no need to use `if user` below (would be good for another use case but not here)